internal/backup_push_handler.go

```diff
@@ -193,7 +193,7 @@ func HandleBackupPush(uploader *Uploader, archiveDirectory string, isPermanent b
 		tracelog.ErrorLogger.FatalError(err)
 	}
 
-	if isPermanent {
+	if isPermanent && previousBackupSentinelDto.BackupStartLSN != nil {
```

Comment:
'previousBackupSentinelDto.BackupStartLSN != nil' means only that previous backup was done by the version which doesn't support increments. We shouldn't enter this branch at all if the current backup is full.